README.md

```diff
@@ -187,18 +192,745 @@ Python 3.8.5 (default, Jan 27 2021, 15:41:15)
+async fn main() -> pyo3::PyResult<()> {
+    pyo3_asyncio::testing::main().await
+}
+```
+
+`pytests/test_example.rs` for the `async-std` runtime:
+```rust
+#[pyo3_asyncio::async_std::main]
+async fn main() -> pyo3::PyResult<()> {
+    pyo3_asyncio::testing::main().await
+}
+```
+
+### Cargo Configuration
+Next, we need to add our test file to the Cargo manifest by adding the following section to the
+`Cargo.toml`
+
+```toml
+[[test]]
+name = "test_example"
+path = "pytests/test_example.rs"
+harness = false
+```
+
+Also add the `testing` and `attributes` features to the `pyo3-asyncio` dependency and select your preferred runtime:
+
+```toml
+pyo3-asyncio = { version = "0.13", features = ["testing", "attributes", "async-std-runtime"] }
+```
+
+At this point, you should be able to run the test via `cargo test`
+
+### Adding Tests to the PyO3 Asyncio Test Harness
+
+We can add tests anywhere in the test crate with the runtime's corresponding `#[test]` attribute:
+
+For `async-std` use the [`pyo3_asyncio::async_std::test`](https://docs.rs/pyo3-asyncio/latest/pyo3_asyncio/async_std/attr.test.html) attribute:
+```rust
+mod tests {
+    use std::{time::Duration, thread};
+
+    use pyo3::prelude::*;
+
+    // tests can be async
+    #[pyo3_asyncio::async_std::test]
+    async fn test_async_sleep() -> PyResult<()> {
+        async_std::task::sleep(Duration::from_secs(1)).await;
+        Ok(())
+    }
+
+    // they can also be synchronous
+    #[pyo3_asyncio::async_std::test]
+    fn test_blocking_sleep() -> PyResult<()> {
+        thread::sleep(Duration::from_secs(1));
+        Ok(())
+    }
+}
+
+#[pyo3_asyncio::async_std::main]
+async fn main() -> pyo3::PyResult<()> {
+    pyo3_asyncio::testing::main().await
+}
+```
+
+For `tokio` use the [`pyo3_asyncio::tokio::test`](https://docs.rs/pyo3-asyncio/latest/pyo3_asyncio/tokio/attr.test.html) attribute:
+```rust
+mod tests {
+    use std::{time::Duration, thread};
+
+    use pyo3::prelude::*;
+
+    // tests can be async
+    #[pyo3_asyncio::tokio::test]
+    async fn test_async_sleep() -> PyResult<()> {
+        tokio::time::sleep(Duration::from_secs(1)).await;
+        Ok(())
+    }
+
+    // they can also be synchronous
+    #[pyo3_asyncio::tokio::test]
+    fn test_blocking_sleep() -> PyResult<()> {
+        thread::sleep(Duration::from_secs(1));
+        Ok(())
+    }
+}
+
+#[pyo3_asyncio::tokio::main]
+async fn main() -> pyo3::PyResult<()> {
+    pyo3_asyncio::testing::main().await
+}
+```
+
+## Migrating from 0.13 to 0.14
+
+So what's changed from `v0.13` to `v0.14`? 
+
+Well, a lot actually. There were some pretty major flaws in the initialization behaviour of `v0.13`. While it would have been nicer to address these issues without changing the public API, I decided it'd be better to break some of the old API rather than completely change the underlying behaviour of the existing functions. I realize this is going to be a bit of a headache, so hopefully this section will help you through it.
```

Comment:
Nit: maybe this file should be wrapped to 100 characters

Reply:
Yeah, it probably should. There's quite a few lines that go on forever, but it's always a pain to do this in the README since I don't have a formatter for markdown. I wouldn't be surprised if VSCode has an extension for it though.

Reply:
VSCode has, https://marketplace.visualstudio.com/items?itemName=stkb.rewrap